java/org/contikios/cooja/GUI.java

```diff
@@ -1464,6 +1465,30 @@ protected void process(List<Object> exs) {
             var config = CreateSimDialog.showDialog(cooja, new CreateSimDialog.SimConfig(title, medium,
                     generatedSeed, seed, delay));
             rv = Objects.requireNonNullElse(config, false);
+            // Try to recreate simulation using a different mote type.
+            var availableMoteTypesObjs = cooja.getRegisteredMoteTypes();
+            var availableMoteTypes = new String[availableMoteTypesObjs.size()];
+            for (int i = 0; i < availableMoteTypes.length; i++) {
+              availableMoteTypes[i] = availableMoteTypesObjs.get(i).getName();
+            }
+            for (var elem : simCfg.getChildren("motetype")) {
+              var moteTypeClassName = elem.getText().trim();
+              var newClass = (String) JOptionPane.showInputDialog(Cooja.getTopParentContainer(),
+                      "The simulation is about to load '" + moteTypeClassName + "'\n" +
+                              "You may try to load the simulation using a different mote type.\n",
+                      "Loading mote type", JOptionPane.QUESTION_MESSAGE, null, availableMoteTypes,
+                      moteTypeClassName);
+              if (newClass == null) {
+                rv = false;
+              } else if (!newClass.equals(moteTypeClassName)) {
+                logger.warn("Changing mote type class: " + moteTypeClassName + " -> " + newClass);
+                // Remove the previous mote-interfaces to load the default interfaces.
+                for (var moteInterface : elem.getChildren("moteinterface").toArray(new Element[0])) {
+                  moteInterface.detach();
+                }
```

Comment:
I think this loop can be simplified as `elem.removeChildren("moteinterface");`

Reply:
Yep, fixed!